src/ec2instances/ec2_instance_mapping.py

```diff
@@ -1,12 +1,35 @@
-from typing import Generator, Iterator
+from typing import Any, Dict, Generator, Iterator
 
+from boto3 import resource
 from instances_map_abc.vm_instance_mapping import VmInstanceMappingBase
 from instances_map_abc.vm_instance_proxy import VmInstanceProxy
 
 from .common.session import get_session
 from .ec2_instance_proxy import Ec2InstanceProxy, Ec2RemoteShellProxy
 
 
+class Ec2AllInstancesMapping(VmInstanceMappingBase[VmInstanceProxy]):
```

Comment:
you can remove inheritance as this not exactly implements a mapping

Reply:
Then maybe not naming it Mapping will be better as well.
Something like Ec2AllInstancesData